docs/openapi_spec.json

```diff
@@ -0,0 +1,212 @@
+                      "$ref": "#/components/schemas/Student"
+                    }]
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/v1/teams": {
+      "post": {
+        "tags": ["Teams"],
+        "summary": "Send the names of the iPraktikum teams",
+        "description": "Send a list of all teams/projects that students can work on during this semesters iteration of the iPraktikum course at TUM",
+        "responses": {
+          "200": {
+            "description": "An array of team (names) in JSON format",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                    "items": {
+                      "oneOf": [{
+                      "$ref": "#/components/schemas/Team"
+                    }]
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    },
+    "/v1/allocation": {
+      "get": {
+        "tags": ["Allocation Result"],
+        "summary": "Retrieve the result of the matching/allocation",
+        "description": "After the allocation is concluded, TEASE can report the results (which team each student was assigned to) without needing to repeat/send back all of the information that was initially sent",
+        "responses": {
+          "200": {
+            "description": "An array of tuples with an identifier for the student and the name of the team that they were allocated to",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "array",
+                    "items": {
+                      "oneOf": [{
+                      "$ref": "#/components/schemas/StudentInTeam"
+                    }]
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "Student": {
+        "type": "object",
+        "properties": {
+          "firstName": {
+            "type": "string"
+          },
+          "lastName": {
+            "type": "string"
+          },
+          "email": {
+            "type": "string"
+          },
+          "tumId": {
+            "type": "string"
+          },
+          "gender": {
+            "$ref": "#/components/schemas/Gender"
+          },
+          "nationality": {
+            "type": "string"
+          },
+          "major": {
+            "type": "string"
+          },
+          "semester": {
+            "type": "number"
+          },
+
+          "germanLanguageLevel": {
+            "type": "string"
+          },
+          "englishLanguageLevel": {
+            "type": "string"
+          },
+
+          "iosDev": {
+            "type": "string"
+          },
+          "appStoreLink": {
+            "type": "string"
+          },
+          "iOSDevExplained": {
+            "type": "string"
+          },
```

Comment:
I don't think that these are necessary anymore. We should update that to be more flexible. More on that in the overall comment